### PR TITLE
Optimisation

### PR DESCRIPTION
**Pull request by wadjih-bencheikh18**
closes : https://github.com/zakodium-oss/react-plot/issues/387

### CODE REVIEW

**Comment by targos** [comment]
I don't think that can be the right fix. If this value is not in the dependencies, we could depend on an outdated version of it.
If the problem is that `state.axes` changes too much, the solution should be to prevent that from happening.

**Comment by wadjih-bencheikh18** [comment]
> The solution should be to prevent that from happening.

No problem, going to search for a solution

**Comment by wadjih-bencheikh18** [comment]
@targos 
i found the problem and fix it

**Comment by wadjih-bencheikh18** [comment]
while fixing the problem i found that in time Axis , `tickLabelFormat` is not specified in AxisContext so after merging this PR .
i will fix it

File: stories/spectra/nmr.stories.tsx [hunk 1]
```diff
@@ -49,7 +49,6 @@ export function StackSpectra() {
       xAxis="x"
       yAxis="y"
       lineStyle={{ stroke: color }}
-      yShift={`${i * -7}`}
```
Comment (targos) [inline stories/spectra/nmr.stories.tsx:52]: can we keep the shift and padding as before?

Comment (wadjih-bencheikh18) [reply stories/spectra/nmr.stories.tsx:52]: Yes no probleme, 
i just thought that the correct graph is without shift

Comment (wadjih-bencheikh18) [reply stories/spectra/nmr.stories.tsx]: fixed

File: stories/spectra/nmr.stories.tsx [hunk 2]
```diff
@@ -61,7 +61,8 @@ export function StackSpectra() {
         position="left"
         label="Intensity / arbitrary"
         hidden
-        paddingEnd="50%"
+        paddingStart="10%"
+        paddingEnd="10%"
```
Comment (targos) [inline stories/spectra/nmr.stories.tsx:65]: please also keep the original padding

Comment (wadjih-bencheikh18) [reply stories/spectra/nmr.stories.tsx]: fixed

**Pull request merged**